File: 파이썬/3.스택,큐/17608.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/17608
'''

n = int(input())
l = []
cnt = 1
for i in range(n) :
    m = int(input())
    l.append(m)

l.reverse() 
max = l[0] 
for i in range(1, n) :
    if max < l[i] :
        max = l[i]
        cnt += 1
print(cnt)

File: C++/07. 스택 - 초급/17608.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/17608
*/

#include <iostream>
#include <stack>

using namespace std;
 
int n, t;
stack<int> s;

int main() {

    cin >> n;
    
    for(int i=0; i<n; i++){
        cin >> t;
        s.push(t);
    }
    
    int cnt = 1;
    int top = s.top();
    s.pop();
    
    while(!s.empty()){
        if(top < s.top()){
            top = s.top();
            cnt++;
        }
        s.pop();
    }
    
    cout << cnt;
    
    return 0;
}